pkg/collector/python/check.go

```diff
@@ -308,8 +321,15 @@ func pythonCheckFinalizer(c *PythonCheck) {
 	// Run in a separate goroutine because acquiring the python lock might take some time,
 	// and we're in a finalizer
 	go func(c *PythonCheck) {
-		glock := newStickyLock() // acquire lock to call DecRef
+		log.Debugf("Running finalizer for check %s", string(c.id))
+
+		glock, err := newStickyLock() // acquire lock to call DecRef
+		if err != nil {
+			log.Warnf("Could not finalize check %s: %s", string(c.id), err.Error())
+			return
```

Comment:
Will fix. it's somewhat of a tricky bigger question though since the type of `c.id` is `ID` but that's just an alias for `string` and you see both types of usages in this file. We probably should standardize on one or the other.